automation_tools/satellite6/capsule.py

```diff
@@ -0,0 +1,398 @@
+            '--certs-tar {}'.format(capsule_hostname, cert_path))
+    return cert_path
+
+
+@task
+def register_capsule():
+    """Register the capsule on the Satellite 6 server."""
+    config = _get_config()
+    run(
+        'rpm -Uvh http://{}/pub/katello-ca-consumer-latest.noarch.rpm'
+        .format(config.server.hostname),
+        warn_only=True
+    )
+    if config.activation_key:
+        run(
+            'subscription-manager register --org={} --activationkey={} --force'
+            .format(config.organization_label, config.activation_key)
+        )
+    elif config.content_view:
+        run(
+            'subscription-manager register --username {} --auto-attach '
+            '--force --password {} --org {} --environment {} '
+            .format(
+                config.admin_user,
+                config.admin_password,
+                config.organization_label,
+                config.content_view,
+            )
+        )
+    else:
+        raise ImproperlyConfigured(
+            'An activation key or content_view name is required.')
+    run('yum repolist')
+
+
+@task
+def capsule_installer(
+        foreman_oauth_key, foreman_oauth_secret, pulp_oauth_secret):
+    """Install and run capsule-installer."""
+    config = _get_config()
+    run('yum -y install capsule-installer')
+    run(
+        'capsule-installer -v --certs-tar *-certs.tar --parent-fqdn {} '
+        '--pulp true --pulp-oauth-secret {} --puppet true --puppetca true '
+        '--foreman-oauth-secret {} --foreman-oauth-key {} '
+        '--register-in-foreman true --qpid-router true --reverse-proxy true'
+        .format(
+            config.server.hostname,
+            pulp_oauth_secret,
+            foreman_oauth_secret,
+            foreman_oauth_key
+        )
+    )
+
+
+@task
+def hammer(command):
+    """Run hammer -u <admin_user> -p <admin_password> --output json <command>.
+
+    :param str command: The hammer subcommand to run.
+    :return: Return a JSON decoded object containing the result of the command.
+        The returned object will exhibit ``failed`` and ``succeeded`` boolean
+        attributes specifying whether the command failed or succeeded, and will
+        also include the return code as the ``return_code`` attribute.
+
+    """
+    config = _get_config()
+    command_result = run(
+        'hammer --username {} --password {} --output json {}'
+        .format(config.admin_user, config.admin_password, command),
+        quiet=True
+    )
+    result = _lower_dict_keys(json.loads(command_result))
+    print(result)
+    if isinstance(result, list):
+        result = _AttributeList(result)
+    elif isinstance(result, dict):
+        result = _AttributeDict(result)
+    result.succeeded = command_result.succeeded
+    result.failed = command_result.failed
+    result.return_code = command_result.return_code
+    return result
+
+
+@task
+def hammer_capsule_lcenvs(capsule_id):
```

Comment:
Maybe. The current name was choose because this is a wrapper around the hammer task. With that said was decided to use hammer prefix instead of get.

I don't have a personal preference here, I can update if that will really make more sense.